src/main/java/com/fasterxml/jackson/databind/EnumNamingStrategies.java

```diff
@@ -43,87 +134,338 @@ private EnumNamingStrategies() {}
+     * <li>"__username" is converted into "USERNAME"</li>
+     * <li>"username" is converted into "USERNAME"</li>
+     * <li>"Username" is converted into "USERNAME"</li>
+     * </ul>
+     *
+     * @since 2.19
+     */
+    public static class UpperSnakeCaseStrategy extends SnakeCaseStrategy {
+
+        /**
+         * An instance of {@link UpperSnakeCaseStrategy} for reuse.
+         *
+         * @since 2.19
+         */
+        public static final UpperSnakeCaseStrategy INSTANCE = new UpperSnakeCaseStrategy();
+
+        @Override
+        public String convertEnumToExternalName(String enumName) {
+            return PropertyNamingStrategies.UpperSnakeCaseStrategy.INSTANCE.translate(toBeanName(enumName));
+        }
+    }
+
+    /**
+     * <p>
+     * An implementation of {@link EnumNamingStrategy} that converts enum names in the typical upper
+     * snake case format to upper camel case format.
+     * This implementation first normalizes to lower camel case using (see {@link LowerCamelCaseStrategy} for details)
+     * and then uses {@link PropertyNamingStrategies.LowerCaseStrategy} to finish converting the name.
+     * <p>
+     * WARNING: Naming conversion conflicts caused by underscore usage should be handled by client.
+     * e.g. Both <code>PEANUT_BUTTER</code>, <code>PEANUT__BUTTER</code> are converted into "peanutbutter".
+     * And "peanutbutter" will be deserialized into enum with smaller <code>Enum.ordinal()</code> value.
+     *
+     * <p>
+     * This results in the following example conversions:
+     * <ul>
+     * <li>"USER_NAME" is converted into "username"</li>
+     * <li>"USER______NAME" is converted into "username"</li>
+     * <li>"USERNAME" is converted into "username"</li>
+     * <li>"User__Name" is converted into "username"</li>
+     * <li>"_user_name" is converted into "username"</li>
+     * <li>"_user_name_s" is converted into "usernames"</li>
+     * <li>"__Username" is converted into "username"</li>
+     * <li>"__username" is converted into "username"</li>
+     * <li>"username" is converted into "username"</li>
+     * <li>"Username" is converted into "username"</li>
+     * </ul>
+     *
+     * @since 2.19
+     */
+    public static class LowerCaseStrategy implements EnumNamingStrategy {
+
+        /**
+         * An instance of {@link LowerCaseStrategy} for reuse.
+         *
+         * @since 2.19
+         */
+        public static final LowerCaseStrategy INSTANCE = new LowerCaseStrategy();
+
+        @Override
+        public String convertEnumToExternalName(String enumName) {
+            return PropertyNamingStrategies.LowerCaseStrategy.INSTANCE.translate(toBeanName(enumName));
         }
+    }
 
-        private static String toLowerCase(String string) {
-            int length = string.length();
-            StringBuilder builder = new StringBuilder(length);
-            for (int i = 0; i < length; i++) {
-                builder.append(charToLowerCaseIfUpper(string.charAt(i)));
```

Comment:
Previous implementation had this strange way of converting to lowercase, but I noticed that the PropertyNamingStrategy implementations are perfectly happy to just use the standard toLowerCase function.

The two approaches handle special characters differently, but I don't think that applies to enum names.